amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/HierarchicalShardSyncer.java

```diff
@@ -140,8 +152,10 @@ public synchronized void checkAndCreateLeaseForNewShards(@NonNull final ShardDet
         }
         final List<Lease> trackedLeases = new ArrayList<>(currentLeases);
         trackedLeases.addAll(newLeasesToCreate);
-        cleanupGarbageLeases(shardDetector, latestShards, trackedLeases, leaseRefresher, multiStreamArgs);
-        if (cleanupLeasesOfCompletedShards) {
+        if (!isLeaseTableEmpty && garbageCollectLeases) {
```

Comment:
To reiterate, we should choose one mode at the beginning of the execution, it can be either empty mode or partial mode. It doesn't matter if the lease table state changes while the execution in progress. It will be more helpful for partial lease table impl, if this mode is passed from outside this class.

Reply:
sure, I moved empty lease table info to be passed as a parameter.